memcache_proxy/dtog.go

```diff
@@ -371,6 +379,16 @@ func unpackFirstStorageCmdArgs(args ...[]byte) (key []byte, flags uint32,
 		return
 	}
 	bytes = uint32(bytesUpc)
+	// Make sure we only accept up to 1000000 bytes to read (max value size
+	// for App Engine cache). Other parts of the code will allocate memory
+	// based on this number and this will only fail once we actually hit
+	// App Engine Memcache.
+	if bytes > maxMemcacheValueSize {
+		err = clientError{"bad command line format",
```

Comment:
Is "bad command line format" actually the correct error message, or was it an artifact of the memcache server choking while reading a packet truncated by the server? looks to me like the error should be more directly stating the problem MEMCACHED_E2BIG:
http://docs.libmemcached.org/client_errors/MEMCACHED_E2BIG.html#MEMCACHED_E2BIG . Also, whatever part of the error message does state the actual problem, we need to be sure that part gets surfaced to cloud logging.